### PR TITLE
Drum GPU

### PR DESCRIPTION
**Pull request by jorshi**
Builds on #116 and makes changes for TorchDrum to run on GPU

### CODE REVIEW

File: torchsynth/module.py [hunk 1]
```diff
@@ -376,27 +379,23 @@ class TorchVCO(TorchSynthModule):
...
-        else:
-            # Create initial phase of zeros like the parameters
-            self.phase = nn.Parameter(
-                data=torch.zeros((self.batch_size,)).unsqueeze(1), requires_grad=False
-            )
-
-        assert self.phase.shape[0] == self.batch_size
+        self.phase = nn.Parameter(
+            data=self.get_parameter("initial_phase"), requires_grad=False
+        )
```
Comment (jorshi) [inline torchsynth/module.py:398]: Not sure if there is a better way to do this -- making it a parameter makes sure that it gets transferred to the GPU when the module is transferred

Comment (turian) [reply torchsynth/module.py:398]: hmm no don't think so

we would be tuning initial phase but not the phase if we rerun this several times is my understanding.

so this should just be a like:
self.phase = T(self.get_parameter("initial_phase"), requires_grad, device=device)

See https://towardsdatascience.com/7-tips-for-squeezing-maximum-performance-from-pytorch-ca4a40951259 Construct tensors directly on GPUs

File: torchsynth/module.py [hunk 2]
```diff
@@ -405,8 +423,9 @@ def _forward(self, mod_signal: Signal) -> Signal:
 
         assert (mod_signal >= -1).all() and (mod_signal <= 1).all()
         control_as_frequency = self.make_control_as_frequency(mod_signal)
-        cosine_argument = self.make_argument(control_as_frequency) + self.phase
-        self.phase = cosine_argument[:, -1]
+        cosine_argument = self.make_argument(control_as_frequency)
+        cosine_argument += self.phase.unsqueeze(1)
```
Comment (turian) [inline torchsynth/module.py:427]: Later but not now we should figure out if unsqueeze is a view (fast) or clone (slow)

File: torchsynth/module.py [hunk 3]
```diff
@@ -757,8 +776,10 @@ def __init__(
         # with the mindset that if you are trying to learn to
         # synthesize a sound, you won't be adjusting the note_on_duration.
         # However, this is easily changed if desired.
-        # TODO: Fix this later
-        self.note_on_duration = T([note_on_duration] * synthglobals.batch_size)
+        # TODO: Fix this later -- should it actually be a parameter?
```
Comment (turian) [inline torchsynth/module.py:779]: Yes link to this issue in code https://github.com/turian/torchsynth/issues/117

Comment (turian) [reply torchsynth/module.py:779]: Not only should it be a parameter, it should be a TorchKnob

File: torchsynth/module.py (hunk 1 above)
Comment (jorshi) [reply torchsynth/module.py:398]: the problem is that the parameters haven't been transferred to the GPU by this point. if you call:
```
vco = TorchSineVCO().to(device)
```
the constructor is called and self.phase gets created on the CPU. Then all the parameters get moved onto the GPU on the `to(device)` call, but self.phase stays on the CPU, which is a problem. By making self.phase a nn.Parameter, then it also gets moved to the GPU on `to(device)`

File: torchsynth/module.py (hunk 2 above)
Comment (jorshi) [reply torchsynth/module.py:427]: it's a view: "The returned tensor shares the same underlying data with this tensor."

**Pull request merged**